pkg/autoscaler/k8sclient/k8sclient.go

```diff
@@ -141,7 +141,6 @@ func discoverAPI(client kubernetes.Interface, kindArg string) (kind string, grou
 	for _, resourceList := range resourceLists {
 		for _, res := range resourceList.APIResources {
 			if res.Name == plural {
-				kind = res.Kind
 				groupVersions[resourceList.GroupVersion] = true
```

Comment:
Interesting bug. If the mismatch does happen, wouldn't this `GroupVersion` get mistakenly set to `tap.linkerd.io/XXX` (even if we omit the `kind`)? Should we do extra filtering on API group to make sure?

Reply:
The wrong `tap.linkerd.io/XXX` will be added to the `groupVersions` map together with the correct ones but we already have in place this `findDeploymentPatcher` or `find{Kind}Patcher` funcs which check for the correct groupVersions first and then return the correct groupVersion.

Ref https://github.com/kubernetes-incubator/cluster-proportional-vertical-autoscaler/blob/a16d6a6abd364a2e6ba6a2128f6fb0584e518a5e/pkg/autoscaler/k8sclient/k8sclient.go#L205-L236

If you want I can enhance the check to be 
```golang
			if res.Name == plural && res.Kind == kind {
				groupVersions[resourceList.GroupVersion] = true
			}
```